### PR TITLE
Gunc/merge

### LINKED ISSUE #2867: new module: GUNC/MERGECHECKM
### Is there an existing module for this?

- [X] I have searched for the existing module

### Is there an open PR for this?

- [X] I have searched for existing PRs

### Is there an open issue for this?

- [X] I have searched for existing issues

### Are you going to work on this?

- [X] If I'm planning to work on this module, I added myself to the `Assignees` to facilitate tracking who is working on the module

### PR DESCRIPTION
**Pull request by jfy133**
## PR checklist

Closes #2867 

- [x] This comment contains a description of changes (with reason).
- [x] If you've fixed a bug or added code that should be tested, add tests!
- [x] If you've added a new tool - have you followed the module conventions in the [contribution docs](https://github.com/nf-core/modules/tree/master/.github/CONTRIBUTING.md)
- [x] If necessary, include test data in your PR.
- [x] Remove all TODO statements.
- [x] Emit the `versions.yml` file.
- [x] Follow the naming conventions.
- [x] Follow the parameters requirements.
- [x] Follow the input/output options guidelines.
- [x] Add a resource `label`
- [x] Use BioConda and BioContainers if possible to fulfil software requirements.
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
  - [ ] `PROFILE=docker pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [x] `PROFILE=singularity pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [ ] `PROFILE=conda pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`

### CODE REVIEW

File: tests/modules/nf-core/gunc/mergecheckm/main.nf [hunk 1]
```diff
@@ -0,0 +1,41 @@
+#!/usr/bin/env nextflow
+
+nextflow.enable.dsl = 2
+
+include { CHECKM_LINEAGEWF              } from '../../../../../modules/nf-core/checkm/lineagewf/main.nf'
+include { CHECKM_QA                     } from '../../../../../modules/nf-core/checkm/qa/main.nf'
+include { GUNC_RUN        } from '../../../../../modules/nf-core/gunc/run/main.nf'
+include { GUNC_DOWNLOADDB } from '../../../../../modules/nf-core/gunc/downloaddb/main.nf'
+include { GUNC_MERGECHECKM } from '../../../../../modules/nf-core/gunc/mergecheckm/main.nf'
```
Comment (maxulysse) [inline tests/modules/nf-core/gunc/mergecheckm/main.nf:9]:
```suggestion
include { CHECKM_LINEAGEWF } from '../../../../../modules/nf-core/checkm/lineagewf/main.nf'
include { CHECKM_QA        } from '../../../../../modules/nf-core/checkm/qa/main.nf'
include { GUNC_RUN         } from '../../../../../modules/nf-core/gunc/run/main.nf'
include { GUNC_DOWNLOADDB  } from '../../../../../modules/nf-core/gunc/downloaddb/main.nf'
include { GUNC_MERGECHECKM } from '../../../../../modules/nf-core/gunc/mergecheckm/main.nf'
```

File: tests/modules/nf-core/gunc/mergecheckm/main.nf [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+
+
+
+
```
Comment (maxulysse) [inline tests/modules/nf-core/gunc/mergecheckm/main.nf:13]:
```suggestion

```

File: tests/modules/nf-core/gunc/mergecheckm/nextflow.config [hunk 3]
```diff
@@ -0,0 +1,11 @@
+process {
+
+    publishDir = { "${params.outdir}/${task.process.tokenize(':')[-1].tokenize('_')[0].toLowerCase()}" }
+
+    withName: CHECKM_QA {
+        ext.args = "-o 2 --tab_table"
+        ext.prefix = { "${meta.id}.qa" }
+    }
+
+
```
Comment (maxulysse) [inline tests/modules/nf-core/gunc/mergecheckm/nextflow.config:10]:
```suggestion
    withName: CHECKM_QA {
        ext.args   = "-o 2 --tab_table"
        ext.prefix = { "${meta.id}.qa" }
    }

```

**Review by maxulysse — APPROVED** [review]

**Comment by jfy133** [comment]
Not sure why the CHeckM conda env is failing to build - it works for me locally, will try one more time - if it fails again I will merge as tests passing for the other two

image

**Comment by jfy133** [comment]
Works fine here:

```
PROFILE=conda TMP=~/tmp nf-core modules test gunc/mergecheckm

                                          ,--./,-.
          ___     __   __   __   ___     /,-._.--~\
    |\ | |__  __ /  ` /  \ |__) |__         }  {
    | \| |       \__, \__/ |  \ |___     \`-._,-`-,
                                          `._,._,'

    nf-core/tools version 2.7.2 - https://nf-co.re


INFO     Press enter to use default values (shown in brackets) or type your own responses                                                                                                                       
─────────────────────────────────────────────────────────────────────────────────────────────── gunc/mergecheckm ───────────────────────────────────────────────────────────────────────────────────────────────
INFO     Running pytest for module 'gunc/mergecheckm'                                                                                                                                                           
============================================================================================= test session starts ==============================================================================================
platform linux -- Python 3.9.15, pytest-7.2.0, pluggy-1.0.0
rootdir: /home/jfellows/Documents/git/jfy133/nf-core-modules, configfile: pytest.ini
plugins: workflow-1.6.0
collecting ... 
collected 1425 items                                                                                                                                                                                           

gunc mergecheckm test_gunc_mergecheckm:
        command:   nextflow run ./tests/modules/nf-core/gunc/mergecheckm -entry test_gunc_mergecheckm -c ./tests/config/nextflow.config -c ./tests/modules/nf-core/gunc/mergecheckm/nextflow.config
        directory: /home/jfellows/tmp/pytest_workflow_ycfzxkbq/gunc_mergecheckm_test_gunc_mergecheckm
        stdout:    /home/jfellows/tmp/pytest_workflow_ycfzxkbq/gunc_mergecheckm_test_gunc_mergecheckm/log.out
        stderr:    /home/jfellows/tmp/pytest_workflow_ycfzxkbq/gunc_mergecheckm_test_gunc_mergecheckm/log.err
'gunc mergecheckm test_gunc_mergecheckm' done.

tests/test_versions_yml.py sssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssss [ 12%]
ssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssss [ 26%]
ssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssss [ 40%]
ssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssss [ 54%]
ssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssss.sssssssssssssssssssssssssssssssss [ 68%]
ssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssss [ 82%]
ssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssss [ 96%]
ssssssssssssssssssssssssssssssssssssssssssssssss                                                                                                                                                         [ 99%]
tests/modules/nf-core/gunc/mergecheckm/test.yml ....                                                                                                                                                     [100%] Keeping temporary directories and logs. Use '--kwd' or '--keep-workflow-wd' to disable this behaviour.


=============================================================================================== warnings summary ===============================================================================================
../../../../.conda/envs/nf-core/lib/python3.9/site-packages/pytest_workflow/plugin.py:122: 822 warnings
  /home/jfellows/.conda/envs/nf-core/lib/python3.9/site-packages/pytest_workflow/plugin.py:122: PytestRemovedIn8Warning: The (fspath: py.path.local) argument to YamlFile is deprecated. Please use the (path: pathlib.Path) argument instead.
  See https://docs.pytest.org/en/latest/deprecations.html#fspath-argument-for-node-constructors-replaced-with-pathlib-path
    return YamlFile.from_parent(parent, fspath=path)

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
========================================================================== 5 passed, 1420 skipped, 822 warnings in 862.35s (0:14:22) ===========================================================================
```

Suspect it's something to do with the GHA install, going to merge

**Pull request merged**